discovery/dns/dns.go

```diff
@@ -284,11 +284,10 @@ func askServerForName(name string, queryType uint16, client *dns.Client, servAdd
 	if err == dns.ErrTruncated {
 		if client.Net == "tcp" {
 			return nil, fmt.Errorf("got truncated message on TCP (64kiB limit exceeded?)")
-		}
-		if edns { // Truncated even though EDNS is used
+		} else {
```

Comment:
Codacy says to drop the `else` and outdent the block contents.  This makes sense to me.